mypy/checker.py

```diff
@@ -616,14 +608,22 @@ def is_implicit_any(t: Type) -> bool:
                 for i in range(len(typ.arg_types)):
                     arg_type = typ.arg_types[i]
 
-                    # Refuse covariant parameter type variables
-                    # TODO: check recuresively for inner type variables
-                    if isinstance(arg_type, TypeVarType):
-                        if i > 0:
-                            if arg_type.variance == COVARIANT:
-                                self.fail(messages.FUNCTION_PARAMETER_CANNOT_BE_COVARIANT,
-                                          arg_type)
-                        # FIX: if i == 0 and this is not a method then same as above
+                    ref_type = self.scope.active_class()
+                    if (isinstance(defn, FuncDef) and ref_type is not None and i == 0
```

Comment:
I see the FuncDef check is to fix the lambda case I reported. This makes me think that was a pretty close call -- I found 1 case of that in over 300K LoC! And the problem was because check_func_def() is called for 'def' as well as for 'lambda', and there are various other checks to weed out static methods and to do the right thing for class and instance methods, but somehow a lambda anywhere in a class would also be checked.

(I'm just muttering to myself here, I just think it's interesting to see how time after time we find that there are more combinations of features than we can possibly test.)